docs/proposals/security-context-constraints.md

```diff
@@ -0,0 +1,303 @@
+## Abstract
+
+Security context constraints relate [security contexts](./security_contexts.md) and
+[service accounts](http://docs.k8s.io/design/service_accounts.md).  They are configured for a
+cluster, namespaces, and users within the cluster to govern how a [security context](./security_contexts.md) is established.
+
+## Motivation
+
+Administration of a multi-tenant cluster requires the ability to provide varying sets of permissions
+among the tenants, the infrastructure components, and end users of the system who may themselves be
+administrators within their own isolated namespace.
+
+Actors in a cluster may include infrastructure that is managed by administrators, infrastructure
+that is exposed to end users (builds, deployments), the isolated end user namespaces in the cluster, and
+the individual users inside those namespaces.  Infrastructure components that operate on behalf of a
+user (builds, deployments) should be allowed to run at an elevated level of permissions without
+granting the user themselves an elevated set of permissions.
+
+## Goals
+
+1.  Associate [service accounts](http://docs.k8s.io/design/service_accounts.md) and users with
+a set of constraints that dictate how a [security context](./security_contexts.md) is established for a pod.
+1.  Provide the ability for users and infrastructure components to run pods with elevated privileges
+on behalf of another user or within a namespace where privileges are more restrictive.
+1.  Secure the ability to reference elevated permissions or to change the constraints under which
+a user runs.
+
+## Use Cases
+
+Use case 1:
+As an administrator, I can create a namespace for a person that can't create privileged containers
+AND enforces that the UID of the containers is set to a certain value
+
+Use case 2:
+As a cluster operator, an infrastructure component should be able to create a pod with elevated
+privileges in a namespace where regular users cannot create pods with these privileges or execute
```

Comment:
As a cluster admin, why would I need to run a pod as privileged in a namespace that doesn't allow privileged pods rather than creating a new namespace that allows privileged pods?